influxql/select.go

```diff
@@ -971,13 +971,13 @@ func (b *exprIteratorBuilder) buildBinaryExprIterator(expr *BinaryExpr) (Iterato
 			return nil, fmt.Errorf("unable to construct an iterator from two literals: LHS: %T, RHS: %T", lhs, rhs)
 		}
 
-		lhs, err := buildExprIterator(expr.LHS, b.ic, b.sources, b.opt, IsSelector(expr.LHS))
+		lhs, err := buildExprIterator(expr.LHS, b.ic, b.sources, b.opt, b.selector)
 		if err != nil {
 			return nil, err
 		}
 		return buildRHSTransformIterator(lhs, rhs, expr.Op, b.opt)
 	} else if lhs, ok := expr.LHS.(Literal); ok {
-		rhs, err := buildExprIterator(expr.RHS, b.ic, b.sources, b.opt, IsSelector(expr.RHS))
+		rhs, err := buildExprIterator(expr.RHS, b.ic, b.sources, b.opt, b.selector)
```

Comment:
`IsSelector` wasn't broken here. It was just improperly used. `IsSelector` correctly identified the expression was a selector, but it was supposed to recognize an earlier conclusion by the query engine not to treat the selectors as selectors.

The reason is because `max()`, when it's treated as a selector, uses the time of the point it selects. When it's treated as an aggregate, it normalizes the time to be the beginning of the current interval. When we deal with any expression where there are multiple selectors (including binary expressions or separately selected), it's not supposed to treat the selectors as selectors.